contracts/StablecoinConverter.sol

```diff
@@ -155,16 +165,17 @@ contract StablecoinConverter is EpochTokenLocker {
                 currentPrices[order.buyToken],
                 currentPrices[order.sellToken]
             );
+            executedBuyAmount = uint128(executedBuyAmount.mul(feeDenominator - 1) / feeDenominator);
```

Comment:
Cool, I will add this comment to the code base.

>(*) Is there any reason we are not using this formula?
yeah, we don't have decimal numbers, so we represent 0.999 as *999/1000

>Also, do we want to handle the case where fee is 0? 
Yes, it would make the unit tests much simpler, but then the code in the smart contract would be more complex. I decided for complexity in the tests over complexity in the smart contract itself

Reply:
If this logic becomes part of getExecutedBuyAmount I think it's worth adding this complexity since it's only in a single case.

Reply:
I recall a time in the past where we initially opted for less contract complexity (namely how the zeroth slot is treated) and eventually landed on moving the complexity into the contract to make the end-to-end tests more coherent. Is this something we expect to change here in the future as well?